### PR TITLE
Bug/gse 1139 show worker boundaries in editor

### PR DESCRIPTION
**Pull request by Vix2021**
#### Description
Fixed bug with editor boundaries not updating instantly when changing nested properties e.g. changing the number of rows of the LBS
Fixed memory leak with multiple multi worker settings classes existing in editor memory.
Fixed memory leak with multiple load balancing strategies existing in editor memory.
Remove restriction to only show worker regions if there is more than one to be consistent with the runtime Spatial Debugger. 

image

image

#### Tests
Added tests to: https://docs.google.com/spreadsheets/d/1UjILAHvCmtM1vNQYWfN6He3nklSHzJ02Ei4b5BbyNfM/edit#gid=0

### CODE REVIEW

**Review by alastairdglennie — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/SpatialMultiWorkerSettings.cpp [hunk 1]
```diff
@@ -27,13 +28,14 @@ void UAbstractSpatialMultiWorkerSettings::PostEditChangeProperty(struct FPropert
 		ValidateNoActorClassesDuplicatedAmongLayers();
 		ValidateAllLayersHaveUniqueNonemptyNames();
 		ValidateAllLayersHaveLoadBalancingStrategy();
+		ASpatialWorldSettings::EditorRefreshSpatialDebugger();
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/SpatialMultiWorkerSettings.cpp:31]: This occurs when the layer info is changes, which as far as I knew, didn't have any visual impact on the spatial debugger. Why are we refreshing it here?

Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/SpatialMultiWorkerSettings.cpp:31]: The SpatialMultiWorkerSettings has a nested "Load Balance Strategy", if this changes AND if this is the current SpatialMultiWorkerSettings selected in the "World Settings" then this needs to update the visualisation. Similarly, if the "Worker Layer" is deleted this change will also need to be reflected.
I will add a check to see if this is the current SpatialMultiWorkerSettings before doing the refresh.
Similarly, I will add a check when the rows, cols etc change to check if the LoadBalancingStrategy is the current one before doing the refresh.

Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/SpatialMultiWorkerSettings.cpp]: Checks added.

File: SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/GridBasedLBStrategy.cpp [hunk 2]
```diff
@@ -186,3 +188,42 @@ UGridBasedLBStrategy::LBStrategyRegions UGridBasedLBStrategy::GetLBStrategyRegio
...
+			const UAbstractSpatialMultiWorkerSettings* MultiWorkerSettings =
+				WorldSettings->MultiWorkerSettingsClass->GetDefaultObject<UAbstractSpatialMultiWorkerSettings>();
+
+			for (const FLayerInfo WorkerLayer : MultiWorkerSettings->WorkerLayers)
+			{
+				if (WorkerLayer.Name == SpatialConstants::DefaultLayer)
+				{
+					const TSubclassOf<UAbstractLBStrategy> VisibleLoadBalanceStrategy = WorkerLayer.LoadBalanceStrategy;
+
+					if (VisibleLoadBalanceStrategy != nullptr && VisibleLoadBalanceStrategy == this->GetClass())
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/GridBasedLBStrategy.cpp:220]:
```suggestion
					if (VisibleLoadBalanceStrategy != nullptr && VisibleLoadBalanceStrategy == GetClass())
```

File: SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/GridBasedLBStrategy.cpp [hunk 3]
```diff
@@ -186,3 +188,42 @@ UGridBasedLBStrategy::LBStrategyRegions UGridBasedLBStrategy::GetLBStrategyRegio
...
+					{
+						ASpatialWorldSettings::EditorRefreshSpatialDebugger();
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/GridBasedLBStrategy.cpp:222]:
```suggestion
						ASpatialWorldSettings::EditorRefreshSpatialDebugger();
						break;
```

File: SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/SpatialMultiWorkerSettings.cpp [hunk 4]
```diff
@@ -27,13 +28,30 @@ void UAbstractSpatialMultiWorkerSettings::PostEditChangeProperty(struct FPropert
...
+{
+	const UWorld* World = GEditor->GetEditorWorldContext().World();
+	check(World != nullptr);
+
+	const ASpatialWorldSettings* WorldSettings = Cast<ASpatialWorldSettings>(World->GetWorldSettings());
+	check(WorldSettings != nullptr);
+
+	const TSubclassOf<USpatialMultiWorkerSettings> VisibleMultiWorkerSettingsClass = WorldSettings->MultiWorkerSettingsClass;
+
+	if (VisibleMultiWorkerSettingsClass != nullptr && VisibleMultiWorkerSettingsClass == this->GetClass())
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/SpatialMultiWorkerSettings.cpp:49]:
```suggestion
	if (VisibleMultiWorkerSettingsClass != nullptr && VisibleMultiWorkerSettingsClass == GetClass())
```

**Review by m-samiec — APPROVED** [review]

**Pull request merged**